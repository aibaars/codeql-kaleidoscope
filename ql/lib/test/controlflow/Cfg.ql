/**
 * @kind graph
 * @id kd/controlflow/cfg
 */

private import codeql.kaleidoscope.Cfg::TestOutput

class MyRelevantNode extends RelevantNode {
  MyRelevantNode() { exists(this) }
}
