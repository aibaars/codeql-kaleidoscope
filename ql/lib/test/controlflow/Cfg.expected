source.kd:
#    1| ...
#-----|  -> def ... = ...

#    1| enter ...
#-----|  -> ...

#    1| exit ...

#    1| exit ... (normal)
#-----|  -> exit ...

#    3| def ... = ...
#-----|  -> def ... = ...

#    3| enter def ... = ...
#-----|  -> y

#    3| exit def ... = ...

#    3| exit def ... = ... (normal)
#-----|  -> exit def ... = ...

#    3| y
#-----|  -> exit def ... = ... (normal)

#    6| def ... = ...
#-----|  -> def ... = ...

#    6| enter def ... = ...
#-----|  -> x

#    6| exit def ... = ...

#    6| exit def ... = ... (normal)
#-----|  -> exit def ... = ...

#    7| if ... then ... else ...
#-----|  -> exit def ... = ... (normal)

#    7| (...)
#-----| true -> 1
#-----| false -> x

#    7| x
#-----|  -> 3

#    7| ...<...
#-----|  -> (...)

#    7| 3
#-----|  -> ...<...

#    8| 1
#-----|  -> if ... then ... else ...

#   10| fib(...)
#-----|  -> x

#   10| ...+...
#-----|  -> if ... then ... else ...

#   10| x
#-----|  -> 1

#   10| ...-...
#-----|  -> fib(...)

#   10| 1
#-----|  -> ...-...

#   10| fib(...)
#-----|  -> ...+...

#   10| x
#-----|  -> 2

#   10| ...-...
#-----|  -> fib(...)

#   10| 2
#-----|  -> ...-...

#   13| def ... = ...
#-----|  -> 10

#   13| enter def ... = ...
#-----|  -> 1

#   13| exit def ... = ...

#   13| exit def ... = ... (normal)
#-----|  -> exit def ... = ...

#   14| var ... in ...
#-----|  -> exit def ... = ... (normal)

#   14| ... = ...
#-----|  -> 1

#   14| 1
#-----|  -> ... = ...

#   14| ... = ...
#-----|  -> ... = ...

#   14| 1
#-----|  -> ... = ...

#   14| ... = ...
#-----|  -> 3

#   15| (...)
#-----|  -> b

#   15| ...:...
#-----|  -> var ... in ...

#   15| for ... in ...
#-----|  -> (...)

#   15| ... = ...
#-----|  -> c

#   15| 3
#-----|  -> ... = ...

#   15| i
#-----|  -> x

#   15| ...<...
#-----| false -> for ... in ...
#-----| true -> c

#   15| x
#-----|  -> ...<...

#   15| 1.0
#-----|  -> i

#   16| c
#-----|  -> a

#   16| ...=...
#-----|  -> a

#   16| ...:...
#-----|  -> b

#   16| ...:...
#-----|  -> 1.0

#   16| a
#-----|  -> b

#   16| ...+...
#-----|  -> ...=...

#   16| b
#-----|  -> ...+...

#   17| a
#-----|  -> b

#   17| ...=...
#-----|  -> ...:...

#   17| b
#-----|  -> ...=...

#   18| b
#-----|  -> c

#   18| ...=...
#-----|  -> ...:...

#   18| c
#-----|  -> ...=...

#   19| b
#-----|  -> ...:...

#   22| fibi(...)
#-----|  -> exit ... (normal)

#   22| 10
#-----|  -> fibi(...)
